samples/hello_events.c

```diff
@@ -115,12 +131,117 @@ static fpga_result inject_ras_fatal_error(fpga_token token, uint8_t err)
 	return result;
 }
 
+/*
+ * Parse command line arguments
+ */
+#define GETOPT_STRING "B:"
+fpga_result parse_args(int argc, char *argv[])
+{
+	struct option longopts[] = {
+		{"bus", required_argument, NULL, 'B'},
+		{NULL, 0, NULL, 0}
+	};
+	
+	int getopt_ret;
+	int option_index;
+	char *endptr = NULL;
+
+	while (-1 != (getopt_ret = getopt_long(argc, argv, GETOPT_STRING, longopts, &option_index))){
+		const char *tmp_optarg = optarg;
+		/* checks to see if optarg is null and if not goes to value of optarg */
+		if ((optarg) && ('=' == *tmp_optarg)){
+			++tmp_optarg;
+		}
+		
+	switch (getopt_ret) {
+	case 'B': /* bus */
+		if (NULL == tmp_optarg)
+			return FPGA_EXCEPTION;
+		endptr = NULL;
+		config.target.bus = (int) strtoul(tmp_optarg, &endptr, 0);
+		if (endptr != tmp_optarg + strnlen(tmp_optarg, 100)) {
+			fprintf(stderr, "invalid bus: %s\n", tmp_optarg);
+			return FPGA_EXCEPTION;
+		}
+		break;
+	
+	default: /* invalid option */
+		fprintf(stderr, "Invalid cmdline options\n");
+		return FPGA_EXCEPTION;
+	}
+	}
+	
+	return FPGA_OK;
+}
+
+
+int find_fpga(fpga_token *fpga, uint32_t *num_matches)
+{
+	fpga_properties filter = NULL;
+	fpga_result 	res;
+
+	/* Get number of FPGAs in system*/
+	res = fpgaGetProperties(NULL, &filter);
+	ON_ERR_GOTO(res, out, "creating properties object");
+
+	res = fpgaPropertiesSetObjectType(filter, FPGA_DEVICE);
+	ON_ERR_GOTO(res, out_destroy, "setting interface ID");
+
+
+	if (-1 != config.target.bus) {
+		res = fpgaPropertiesSetBus(filter, config.target.bus);
+		ON_ERR_GOTO(res, out_destroy, "setting bus");
+	}
+		
+	res= fpgaEnumerate(&filter, 1, fpga, 1, num_matches);
+	ON_ERR_GOTO(res, out_destroy, "enumerating FPGAs");
+
+out_destroy:
+	res = fpgaDestroyProperties(&filter);
```

Comment:
If one of the ON_ERR_GOTO statements on 185, 188, 193, or 197 is taken, then the failure code stored in res is overwritten by line 200.  Could you create a 
fpga_result dres;

dres = fpgaDestroyProperties
return (res == FPGA_OK) ? dres : res;